### ISSUE TITLE
narrative: outpatient SOAP Plan section is byte-identical across every visit in 92.1% of patients (Assessment identical in 23.5%, both in 16.7%)

### ISSUE THREAD

**Issue by TomoOkuyama** [TomoOkuyama/clinosim#1180]
## Description

Outpatient SOAP progress notes (`Composition` where `title` is `外来経過記録（SOAP）`) show byte-identical text in the Plan (`section[title='計画']`) and Assessment (`section[title='評価']`) sections across every SOAP note for the same patient, over multi-month follow-up windows.

Concretely, in the 2026-09-07 build:

- **92.1 %** of patients (3,810 of 4,137 patients with ≥3 outpatient SOAP notes) have Plan section text that is byte-identical across every one of their SOAP notes.
- **23.5 %** (972 / 4,137) have Assessment section text byte-identical across every one of their SOAP notes.
- **16.7 %** (692 / 4,137) have both Assessment and Plan byte-identical across every one of their SOAP notes.
- **93.7 %** of individual SOAP notes belonging to a patient with ≥3 SOAP notes (25,429 / 27,148) sit inside a "Plan is a template constant" patient.

Real EHR outpatient SOAP is one of the highest-signal sources of clinical trajectory: dose adjustments, lab trending vs. target, symptom / adherence / lifestyle changes, response to previous plan, next-visit interval. Freezing the Assessment and Plan at their first-visit text and re-emitting the same string on every subsequent visit turns the SOAP series into `N × the same note`.

This is the outpatient counterpart of #1154 (which reports inpatient progress-note temporal invariance of *lab values*). #1154 is about numeric lab values not evolving during a stay; the present issue is about the Assessment and Plan **narrative text itself** being byte-identical across every outpatient visit — a stronger form of invariance and at outpatient scale (27,842 SOAP notes / 4,612 patients).

## Environment

- Generator output: **5,421 patients / 1,954,258 resources / 26 resource types** (2026-09-07 build, clinosim 0.6.0 @ ac297882b8, seed 1294)
- All **27,842** `Composition` resources whose title contains `SOAP` or `経過記録` scanned via bulk FHIR search
- No FHIR server required to reproduce; the analysis runs on the NDJSON files alone

## Scope of the invariance (full 27,842 SOAP composition sweep, aggregated per patient)

| | Patients (of 4,137 with ≥3 SOAP notes) | % |
|---|---:|---:|
| Assessment section byte-identical across every SOAP note | 972 | **23.5 %** |
| Plan section byte-identical across every SOAP note | **3,810** | **92.1 %** |
| Both Assessment **and** Plan byte-identical | 692 | **16.7 %** |

At the note level (of 27,148 SOAP notes belonging to a ≥3-note patient):

| | SOAP notes | % |
|---|---:|---:|
| Sitting in a patient whose Assessment is 100 % duplicated | 5,560 | 20.5 % |
| Sitting in a patient whose Plan is 100 % duplicated | **25,429** | **93.7 %** |

The Plan-invariance signal is population-wide, not a long-tail artifact.

### Sample 1 — 7 SOAP notes over 7 months, **both Assessment and Plan byte-identical**

Patient `pt-12fee9377f56` (male, born 1945-02-04) — 7 outpatient SOAP notes on 2025-11-01, 2026-01-18, 2026-02-08, 2026-03-25, 2026-04-25, 2026-05-12, 2026-06-04 — 7 visits spanning ~7 months at 1–2 month spacing.

Each of the 7 SOAP notes contains the following Assessment section, character-for-character:

```
脂質異常症のフォローアップ。LDL 目標値を心血管リスク層別で評価。
1. 脂質異常症: 本日測定なし、次回受診時再評価。
2. 心不全: 本日測定なし、次回受診時再評価。
3. 心房細動・心房粗動: 本日測定なし、次回受診時再評価。
4. 胃食道逆流症: 本日測定なし、次回受診時再評価。
5. 背部痛（腰痛を含む）: 本日測定なし、次回受診時再評価。
6. 肝及び肝内胆管の悪性新生物＜腫瘍＞: 本日測定なし、次回受診時再評価。
7. アルコール使用障害: 本日測定なし、次回受診時再評価。
```

And the following Plan section, character-for-character:

```
スタチン継続、必要に応じて用量調整。食事療法 (飽和脂肪酸・糖質制限)
と運動療法を再指導。次回外来は 3 か月後、AST・ALT・CK・脂質パネル
を採血。
```

Note the internal contradiction: the Plan says `次回外来は 3 か月後` ("next visit in 3 months") but the actual next-visit interval implied by the generated series is 1–2 months. Even the recommended follow-up cadence never varies to match the patient's actual visit rhythm.

### Sample 2 — 7 SOAP notes, Assessment and Plan byte-identical

Patient `pt-becccc37469e` (female) — 7 outpatient SOAP notes; the SHA-256 hash of the Assessment section text is the same on all 7, and the Plan hash is the same on all 7.

### Sample 3 — 8 SOAP notes, Plan byte-identical on all 8

Patient `pt-291dea191fab` (male) — 8 outpatient SOAP notes; Assessment has 6 distinct hash values (some varied text between visits), but the Plan hash is one single value across all 8 notes. Plan invariance persists even when Assessment does drift.

## Expected behavior

An outpatient SOAP note is per-visit. Between visits, at least one of the following must move: measured labs on the day, medication dose or adherence, symptom vs. previous visit, lifestyle counseling checkpoint, next follow-up interval. Generation should reflect at least the observable per-visit deltas:

- **Assessment** should reference the labs / vitals actually attached to that visit's `Encounter` (see #1155 for the related "no findings" fallback) rather than a static problem-list dump.
- **Plan** should update at least one field per visit — the medication line, the follow-up interval, or the next-labs list — driven by what changed since the previous visit (dose titration, out-of-range lab, patient-reported adherence).

If the generator has no per-visit signal to condition on and defaults to a template, at minimum it should perturb the interval / lab-panel list from visit to visit so the Plan hash varies, and should emit a `note` such as `"template-plan"` on the section so downstream consumers can filter these out. Emitting the same byte string on 7 consecutive visits is the failure mode this issue reports.

## Reproduction

Runs against the generator's NDJSON output; no server required. Requires only `Composition.ndjson`.

```python
import json, re, hashlib
from pathlib import Path
from collections import defaultdict

DIR = Path("./fhir_r4")  # adjust to your generator output directory

def clean(div):
    return re.sub(r"\s+", " ", re.sub(r"<[^>]+>", "", div)).strip()

# patient -> list of (assess_hash, plan_hash)
notes_by_pt = defaultdict(list)
for line in (DIR / "Composition.ndjson").open():
    r = json.loads(line)
    title = r.get("title", "")
    if "SOAP" not in title and "経過記録" not in title:
        continue
    pid = ((r.get("subject") or {}).get("reference") or "").replace("Patient/", "")
    assess = plan = ""
    for s in r.get("section", []):
        div = clean((s.get("text") or {}).get("div", ""))
        if s.get("title") == "評価": assess = div
        if s.get("title") == "計画": plan  = div
    notes_by_pt[pid].append((
        hashlib.md5(assess.encode()).hexdigest()[:12],
        hashlib.md5(plan.encode()).hexdigest()[:12],
    ))

pt_ge3 = {pid: hs for pid, hs in notes_by_pt.items() if len(hs) >= 3}
identical_assess = sum(1 for hs in pt_ge3.values() if len({h[0] for h in hs}) == 1)
identical_plan   = sum(1 for hs in pt_ge3.values() if len({h[1] for h in hs}) == 1)
identical_both   = sum(1 for hs in pt_ge3.values() if len({h[0] for h in hs}) == 1 and len({h[1] for h in hs}) == 1)

print(f"total SOAP notes: {sum(len(v) for v in notes_by_pt.values())}")
print(f"patients with >=3 SOAP notes: {len(pt_ge3)}")
print(f"  assessment 100% identical across all their notes: {identical_assess} ({identical_assess/len(pt_ge3)*100:.1f}%)")
print(f"  plan       100% identical: {identical_plan} ({identical_plan/len(pt_ge3)*100:.1f}%)")
print(f"  both assess+plan 100% identical: {identical_both} ({identical_both/len(pt_ge3)*100:.1f}%)")
```

Expected output on the 2026-09-07 build (5,421 patients / seed 1294):

```
total SOAP notes: 27842
patients with >=3 SOAP notes: 4137
  assessment 100% identical across all their notes: 972 (23.5%)
  plan       100% identical: 3810 (92.1%)
  both assess+plan 100% identical: 692 (16.7%)
```

## Impact

- **Clinical trajectory is invisible in the outpatient record.** For 92.1 % of patients the Plan never evolves. A reader looking at the SOAP series has no way to see dose titration, follow-up-interval change, or which lab the clinician chose to reorder — all information a real EHR carries.
- **Chart-view UIs render N identical notes.** A patient-timeline showing a stack of SOAP note cards will display the same paragraph 7 times in a row. There is nothing for the reader to look at.
- **The Plan contradicts the actual visit cadence.** In Sample 1 the Plan says `次回外来は 3 か月後` on every one of 7 visits that are actually 1–2 months apart. Consumers that trust the narrative for follow-up scheduling will schedule against a value the generator itself contradicts through the timing of the next `Composition`.
- **LLM/RAG training data is degraded.** Fine-tuning a summarizer or a clinical-reasoning model on 27,842 SOAP notes that collapse to a small number of duplicated Plan templates teaches "the plan for outpatient SOAP is a constant per patient" — the exact anti-pattern the training set is meant to avoid.
- **Not fixable downstream.** Consumers cannot re-generate what the Plan should have been from `Encounter` + `Observation` + `MedicationRequest` alone without redoing the LLM generation the template is supposed to be doing.

Distinct from #1154, which reports inpatient progress-note *lab-value* invariance within one admission. The present issue is at outpatient scale (27,842 notes, 92 % of patients) and is about the Plan **text itself** being byte-identical across every visit, not just its lab-value slot.
